include/deal.II/grid/grid_out.h

```diff
@@ -1080,6 +1080,39 @@ class GridOut
                   std::ostream                      &out) const;
 
   /**
+   * Write triangulation in VTU format for each processor, then combine these
+   * files into a .pvtu file. The output is in the form
+   * <tt>filename_without_extension.proc000*.vtu</tt> where * is 0,1,...,n_proc-1 and
+   * <tt>filename_without_extension.pvtu</tt>. The input <tt>view_levels</tt> can be
+   * set as true to view each level of a multilevel method. The input
+   * <tt>include_artificial</tt> can be set as true to view the artificial cells for
+   * each processor. Each .vtu and .pvtu file will have the attributes subdomain,
+   * lvlsubdomain, level, and proc_writing. The level value can be used to seperate the
+   * image into the view of the grid on each level of a multilevel method and the
+   * proc_writing value can be used to seperate the image into each processors owned and
+   * ghost cells.
+   * This is accomplished by applying the "warp by scalar" filter in paraview
+   * to each of the values. After opening the .pvtu file of a mesh where the input
+   * <tt>view_levels</tt> is set to true, select the "warp by scalar"
+   * filter. For the "Scalars" input select <tt>proc_writing</tt> and for the "Normal" input
+   * enter in 1 0 0, then click Apply. Next select the "warp by scalar" filter again. For the
+   * "Scalars" input select <tt>level</tt> and for the "Normal" input enter in 0 1 0,
+   * then click Apply. This will give you the following image.
+   * @image html write_debug_vtu_levels.png
+   * If the <tt>view_levels</tt> remains at false, thereby only giving the mesh for the active
+   * level, it is enough to seperate the image into the views written by different processors.
+   * This is shown in the following image where the <tt>include_artificial</tt> input is set as true.
+   * @image html write_debug_vtu_active.png
+   * Note: Depending on the size of the mesh you may need to increase the "Scale Factor" input
+   * so that each piece does not overlap.
+   */
+  template <int dim, int spacedim>
+  void write_debug_vtu_per_processor (const Triangulation<dim,spacedim> &tria,
```

Comment:
I'm not enamored with the name. "debug" means that the function may only be useful to *debug* something, but it may also be useful to *understand*, to *document*, or just *visualize* a mesh hierarchy. How about just `write_mesh_per_processor_as_vtu()`?